tests/testStatsdClient.cpp

```diff
@@ -1,18 +1,49 @@
 #include <StatsdClient.hpp>
-#include <cassert>
+#include <StatsdServer.hpp>
 #include <iostream>
 
 using namespace Statsd;
 
+// We just keep storing metrics in an vector until we hear a special one, then we bail
```

Comment:
so basically we spawn a thread in the background which makes a server to listen to our client. everytime the server gets a message it parses it (splitting it on endline characters) and pushes it into the vector it was provided. once it hears a special message with the keyword `DONE` in it it quits